app/src/main/java/org/dharmaseed/androidapp/PlayTalkActivity.java

```diff
@@ -324,4 +357,151 @@ public void onStopTrackingTouch(SeekBar seekBar) {
+            Drawable icon = ContextCompat.getDrawable(this, R.drawable.ic_file_download_green_24dp);
+            downloadButton.setImageDrawable(icon);
+            downloadButton.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View view) {
+                    onDownloadButtonClicked(view);
+                }
+            });
+        }
+    }
+
+    public void onDownloadButtonClicked(View view) {
+        Log.d(LOG_TAG, "Downloading talk " + talk.getId());
+        downloadTalk();
+    }
+
+    public void onDeleteTalkClicked(View view) {
+        DialogFragment dialog = new DeleteTalkFragment();
+        dialog.show(getFragmentManager(), "DeleteTalkFragment");
+    }
+
+    public void deleteTalk() {
+        if (!TalkManager.deleteTalk(talk)) {
+            showToast("Unable to delete talk.", Toast.LENGTH_SHORT);
+        } else {
+            dbManager.deleteTalk(talk);
+            showToast("Talk deleted.", Toast.LENGTH_SHORT);
+            toggleDownloadImage();
+            Log.d(LOG_TAG, "Deleted talk " + talk.getId());
+        }
+    }
+
+    public void startLoadingAnim() {
+        ImageButton downloadButton = (ImageButton) findViewById(R.id.download_button);
+        Drawable icon = ContextCompat.getDrawable(this, R.drawable.ic_downloading);
+        downloadButton.setImageDrawable(icon);
+        if (icon instanceof Animatable)
+            ((Animatable) icon).start();
+    }
+
+    public void stopLoadingAnim() {
+        Drawable icon = ContextCompat.getDrawable(this, R.drawable.ic_downloading);
+        if (icon instanceof Animatable)
+            ((Animatable) icon).stop();
+    }
+
+    public void showToast(String message, int length) {
+        Toast.makeText(getApplicationContext(), message, length).show();
+    }
+
+    @Override
+    public void onDeleteTalkPositiveClick(DialogFragment dialogFragment) {
+        deleteTalk();
+    }
+
+    @Override
+    public void onDeleteTalkNegativeClick(DialogFragment dialogFragment) {
+        // do nothing
+    }
+
+    class DownloadTalkTask extends AsyncTask<Talk, Integer, Long> {
+
+        private static final String LOG_TAG = "DownloadTalkTask";
+        private Talk talk;
+
+        public DownloadTalkTask() {}
+
+        @Override
+        protected Long doInBackground(Talk... talks) {
+            startLoadingAnim();
+            long totalSize = 0;
+            if (talks.length > 0) {
+                // only download the first one if we get passed multiple
+                this.talk = talks[0];
+                totalSize = TalkManager.download(this.talk);
+            }
+            return totalSize;
+        }
+
+        @Override
+        protected void onPostExecute(Long size) {
+            // TODO change logs to show dialog
+            if (size > 0) {
+                if (dbManager.addDownload(this.talk) == 1) {
+                    showToast("Talk downloaded.", Toast.LENGTH_SHORT);
```

Comment:
I noticed that if I navigate to one talk, start downloading, then quickly navigate to another talk and try to download it, the download does start but the animation doesn't.  I don't really think that's a big deal, and this might not be a common use case anyway, but I wondered whether it would be nice to put the title of the talk in the toast message to tell the user which talks they've successfully downloaded.  This would probably also be nicely addressed by #32.

By the way, I'm pretty sure Dharma Seed rate-limits the number of talks you can download, because all of my downloads suddenly stopped working with 404 errors after I had triggered perhaps 30 or so for testing purposes. :P  I remember hitting similar limits when developing the app a few years back.  Just wanted to let you know in case you hit similar behavior at some point. :)

Reply:
Oops, I was writing code to answer this as you merged.

It looks like AsyncTasks are queued if multiple are launched at the same time. By moving the start of the loading animation into `onPreExecute()`, the animation now starts for every download.

As for the rate-limiting, yes I think you're right. I encountered the same thing, lol.

Reply:
Oops, my apologies.  If you'd like, I can reset master back to its previous state to temporarily, which might allow us to re-use this PR (not totally sure if github supports that, but it can't hurt to try).  Or if you want to make a new PR for your new fix, that works too.  Sorry I jumped the gun! :P

Reply:
No, no worries! I should have mentioned I was working on this. I'll just submit a new PR 😄

Reply:
Cool, sounds good! :)